### ISSUE TITLE
Analytics role not working

### ISSUE THREAD

**Issue by sergiorodriguez82** [aws-amplify/amplify-cli#10869]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Angular

### Amplify APIs

Authentication, Analytics, REST API, GraphQL API, DataStore, Storage

### Amplify Categories

auth, analytics

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

Hi I just added the analytics category to my app and Im getting this error
ERROR Error: Uncaught (in promise): User: arn:aws:sts::XXXXX:assumed-role/us-east-2_qVBo3N2qI-AdminGroupRole/CognitoIdentityCredentials is not authorized to perform: mobiletargeting:PutEvents on resource: arn:aws:mobiletargeting:us-east-1:XXXXXX:apps/4a167a45b9784eba813852e3f3c86205/events
User: arn:aws:sts::704766592508:assumed-role/us-east-2_qVBo3N2qI-AdminGroupRole/CognitoIdentityCredentials is not authorized to perform: mobiletargeting:PutEvents on resource: arn:aws:mobiletargeting:us-east-1:704766592508:apps/4a167a45b9784eba813852e3f3c86205/events

also im getting this warning
AWSPinpointProvider - Please ensure you have updated your Pinpoint IAM Policy with the Action: "mobiletargeting:PutEvents" in order to record events

### Expected behavior

Events being recorded as expected

### Reproduction steps

Add the analytics category
Add the record event as shown in the docs

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by jedwardblack** [comment]
To expand on the solution @ezeev presented, see the following snippet for an example `auth/user-pool-group-precedence.json` file.

```
[
  {
    "groupName": "SuperAdmins",
    "precedence": 1,
    "customPolicies": [
      {
        "PolicyName": "pinpoint-put",
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": [
                "mobiletargeting:UpdateEndpoint",
                "mobiletargeting:PutEvents"
              ],
              "Resource": [
                "arn:aws:mobiletargeting:*:YOU_ORG_ID:apps/YOUR_APP_ID*"
              ]
            }
          ]
        }
      }
    ]
  },
  {
    "groupName": "Admins",
    "precedence": 2,
    "customPolicies": [
      {
        "PolicyName": "pinpoint-put",
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": [
                "mobiletargeting:UpdateEndpoint",
                "mobiletargeting:PutEvents"
              ],
              "Resource": [
                "arn:aws:mobiletargeting:*:YOUR_ORG_ID:apps/YOUR_APP_ID*"
              ]
            }
          ]
        }
      }
    ]
  }
]
```

**Comment by ezeev** [comment]
@jedwardblack Yup! that's how we did it.

**Comment by Jupdi** [comment]
I thought the issue is the same as https://github.com/aws-amplify/amplify-cli/issues/4772  thought was fixed in https://github.com/aws-amplify/amplify-cli/pull/10455, butt seems the bug still exsists.

**Comment by joekiller** [comment]
Agreed doesn't work still. Seems the group role isn't being used by the Analytics plugin

**Comment by joekiller** [comment]
For us, the problem was that we added the "Pre token generation Lambda trigger" so taking that out made it work again.

Looking at it closer, we missed that the [groupOverrideDetails needed to include the requests iam groups](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html#cognito-user-pools-lambda-trigger-syntax-pre-token-generation) so we were clobbering them off. spreading the groupConfiguration over the overrideDetails and then including our updates fixed it up.

![image](https://github.com/aws-amplify/amplify-cli/assets/1022919/6af14eba-c4d8-4968-881c-171b8e99c89e)


<img width="409" alt="Screenshot 2023-08-09 at 7 56 32 PM" src="https://github.com/aws-amplify/amplify-cli/assets/1022919/d549fea3-1bd0-405e-8a58-ea506da2d237">

**Comment by charlieforward9** [comment]
I am [experiencing a similar auth issue when configuring](https://github.com/aws-amplify/amplify-cli/issues/11631#issuecomment-3530398814) Amplify Analytics for Kinesis.

With Pinpoint end of support, I am curious of the urgency the Amplify team is putting behind this package, and if alternatives are recommended instead? 

Otherwise, consolidating effort here to the Kinesis plugin would be reasonable to propose. In this case, #11631 needs attention.

**Comment by charlieforward9** [comment]
> The solution was to add "customPolicies" to the `auth/user-pool-group-precedence.json` file.

This solved my problem. 

Still curious on the status of the Analytics package in general... this issue is pretty old.
